### PR TITLE
Add email_verified_code to bypass 2FA by email

### PR DESCRIPTION
**Pull request by nono**
It will allow the cloudery to inform the stack that it has already checked that the user can receives email on their address.

### CODE REVIEW

File: model/instance/store.go [hunk 1]
```diff
@@ -13,11 +13,16 @@ import (
 // Store is an object to store and retrieve session codes.
 type Store interface {
 	SaveSessionCode(db prefixer.Prefixer, code string) error
+	SaveEmailVerfiedCode(db prefixer.Prefixer, code string) error
 	CheckAndClearSessionCode(db prefixer.Prefixer, code string) bool
+	CheckEmailVerifiedCode(db prefixer.Prefixer, code string) bool
```
[the remark this answers is not in the document]
Comment (nono) [reply model/instance/store.go:18]: Nope, this function doesn't clear the code, we just let it expire after 15 minutes

File: model/instance/store.go [hunk 2]
```diff
@@ -67,29 +72,59 @@ func (s *memStore) cleaner() {
...
+	key := emailVerifiedCodeKey(db, code)
+	s.vals[key] = time.Now().Add(emailVerifiedCodeTTL)
 	return nil
 }
 
 func (s *memStore) CheckAndClearSessionCode(db prefixer.Prefixer, code string) bool {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	exp, ok := s.vals[code]
+	key := sessionCodeKey(db, code)
```
[the remark this answers is not in the document]
Comment (nono) [reply model/instance/store.go:91]: Yes, there is a risk, but it is very low (something like 0,1% than 1 user will be concerned)

[the remark this answers is not in the document]
Comment (taratatach) [reply model/instance/store.go:91]: What would be the consequences? A failed login?

File: model/instance/store.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (taratatach) [reply model/instance/store.go:18]: I went a bit fast and missed the surrounding block with the time condition.

File: model/instance/store.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (nono) [reply model/instance/store.go:91]: Someone will have to reload their webapp in the flagship app (ie go back and retry, or kill the flagship app and restart, as there is no reload button).

**Review by taratatach — APPROVED** [review]

**Pull request merged**